Libraries/pbxbuild/Sources/Tool/SwiftResolver.cpp

```diff
@@ -232,6 +232,73 @@ AppendObjcHeader(
     }
 }
 
+static void
+AppendUnderlyingModule(
+    std::vector<std::string> *args,
+    std::vector<Tool::Invocation::AuxiliaryFile> *auxiliaryFiles,
+    Tool::Context const *toolContext,
+    pbxsetting::Environment const &environment,
+    std::string const &headerName)
+{
+    /* No module map means no underlying module to import. */
+    ext::optional<Tool::ModuleMapInfo::Entry> const &moduleMap = toolContext->moduleMapInfo().moduleMap();
+    if (!moduleMap) {
+        return;
+    }
+
+    /*
+     * Create a module map to import the underlying module. However, exclude the Swift-generated
+     * header here, to avoid having this module try and import its own generated Objective-C structures.
+     */
+    std::string moduleName = environment.resolve("PRODUCT_MODULE_NAME");
+
+    std::string unextendedModuleMap = "\n";
+    unextendedModuleMap += "module " + moduleName + ".__Swift {\n";
+    unextendedModuleMap += "  exclude header \"" + headerName + "\"\n";
+    unextendedModuleMap += "}\n";
+
+    auto unextendedModuleMapData = std::vector<uint8_t>(unextendedModuleMap.begin(), unextendedModuleMap.end());
+    auto unextendedModuleMapChunk = Tool::Invocation::AuxiliaryFile::Chunk::Data(unextendedModuleMapData);
+
+    std::string unextendedModuleMapPath = environment.resolve("TARGET_TEMP_DIR") + "/" + "unextended-module.modulemap";
+    auto unextendedModuleMapFile = Tool::Invocation::AuxiliaryFile(unextendedModuleMapPath, { moduleMap->contents(), unextendedModuleMapChunk });
+    auxiliaryFiles->push_back(unextendedModuleMapFile);
+
+    /*
+     * Create a VFS overlay to load the above module map. This replaces the module map in the product,
+     * which does not contain the extra Swift rule, and may not even exist yet (depending on ordering).
+     */
+    std::string unextendedModuleOverlayPath = environment.resolve("TARGET_TEMP_DIR") + "/" + "unextended-module-overlay.yaml";
+
+    std::string unextendedModuleOverlay;
+    unextendedModuleOverlay += "{\n";
```

Comment:
This is YAML, not JSON, although I think it's valid JSON too so I suppose it could use the `plist` API to write it out. But it's simple enough this seemed easiest?